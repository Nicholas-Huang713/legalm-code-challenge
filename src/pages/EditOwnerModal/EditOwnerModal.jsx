import Modal from "../../components/Modal/Modal";
import Form from "../../components/Form/Form";
import { editOwner } from "../../state/slices/owner/ownerThunks";
import { useDispatch, useSelector } from "react-redux";

export default function EditOwnerModal({ handleCloseEditOwnerModal }) {
  const dispatch = useDispatch();
  const currentDogWithOwnerInfo = useSelector((state) => state.dog.dogs);
  return (
    <Modal onClose={handleCloseEditOwnerModal}>
      <Form
        isEditMode={true}
        sendDataToServer={(data) => dispatch(editOwner(data))}
        handleCloseForm={handleCloseEditOwnerModal}
        currentDogWithOwnerInfo={currentDogWithOwnerInfo}
      />
    </Modal>
  );
}
