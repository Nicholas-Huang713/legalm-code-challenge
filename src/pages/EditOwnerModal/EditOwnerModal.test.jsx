import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import EditOwnerModal from "./EditOwnerModal";
import { dogDataInStore, ownersList } from "../../test/mocks/mockValues";

const thunkMiddleware = require("redux-thunk").thunk;
const mockStore = configureMockStore([thunkMiddleware]);
const mockCloseModal = vi.fn();
let store;

const initStoreWithDogData = () => {
  store = mockStore({
    owner: ownersList,
    dog: dogDataInStore,
  });
  store.dispatch = vi.fn();
};

describe("EditOwnerModal component", () => {
  it("submit calls handleCloseDetailModal", async () => {
    initStoreWithDogData();
    const { container } = render(
      <Provider store={store}>
        <EditOwnerModal handleCloseEditOwnerModal={mockCloseModal} />
      </Provider>
    );
    const nameInput = screen.getByLabelText(/Name/i);
    fireEvent.change(nameInput, {
      target: { value: "John Wick" },
    });
    fireEvent.click(screen.getByText("Submit"));
    await waitFor(() => {
      expect(store.dispatch).toHaveBeenCalledWith(expect.any(Function));
    });
    expect(container).toMatchSnapshot();
  });
});
