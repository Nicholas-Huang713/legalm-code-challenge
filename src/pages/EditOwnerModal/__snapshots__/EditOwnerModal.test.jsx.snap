// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditOwnerModal component > submit calls handleCloseDetailModal 1`] = `
<div>
  <div
    class="modal-backdrop"
    data-testid="modal-backdrop"
  >
    <div
      class="modal-content"
    >
      <button
        class="close-btn"
      >
        X
      </button>
      <section
        class="modal-content-main"
      >
        <form>
          <header>
            <h2>
              Edit Profile
            </h2>
          </header>
          <div
            class="form-input-container"
          >
            <div>
              <label
                for="name"
              >
                Name 
              </label>
              <input
                class="form-input"
                id="name"
                name="name"
                type="text"
                value=""
              />
              <div />
            </div>
            <div>
              <label
                for="exp"
              >
                Yrs of Exp 
              </label>
              <input
                id="exp"
                name="exp"
                type="number"
                value="0"
              />
            </div>
          </div>
          <div>
            <img
              class="form-img"
              src=""
            />
          </div>
          <div
            class="form-dog-info"
          >
            <div>
              <strong>
                Name:
              </strong>
               
            </div>
            <div>
              <strong>
                Favorite food:
              </strong>
               
            </div>
          </div>
          <button
            type="submit"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
  </div>
</div>
`;
